models/src/main/scala/coop/rchain/casper/protocol/CasperMessage.scala

```diff
@@ -356,23 +358,60 @@ object ProcessedDeploy {
   }
 }
 
+final case class SlashSystemDeployData(invalidBlockHash: BlockHash, issuerPublicKey: PublicKey)
+
+final case class SystemDeployData(maybeSlashData: Option[SlashSystemDeployData])
```

Comment:
Why not just have `SlashSystemDeployData` extend `SystemDeployData`? Then `playSucceeded` is subtype polymorphic over the `SystemDeployData`. More extensible.